code/modules/medical/genetics/bioEffects/hemochromia.dm

```diff
@@ -0,0 +1,442 @@
+	icon_state  = "hemochromia_unknown"
+	var/typeRange = 0
+	var/duplicateCheck = 0
+	var/blood_color_R
+	var/blood_color_G
+	var/blood_color_B
+
+
+
+	OnLife()
+		if(ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			if(prob(12))
+				if(!istype(H.loc, /obj/machinery/genetics_scanner))
+					if(H.bioHolder.HasEffectInEither("hemochromia_rust" || "hemochromia_bronze" || "hemochromia_gold" || "hemochromia_lime" || "hemochromia_olive" || "hemochromia_jade" || "hemochromia_teal" || "hemochromia_cobalt" || "hemochromia_indigo" || "hemochromia_purple" || "hemochromia_violet" || "hemochromia_fuschia"))
+						duplicateCheck = 1
+					if(duplicateCheck != 0)
+						holder.RemoveEffect(src.id)
+						holder.RemovePoolEffect(src)
+					if(id == "hemochromia_unknown")
+						if(4 >= rand(13))
+							switch(H.bioHolder.bloodType)
+								if("A+")
+									if(pick(1, 2) == 1)
+										typeRange = 1
+									else
+										typeRange = 2
+								if("A-") typeRange = 3
+								if("B+") typeRange = 4
+								if("B-") typeRange = 5
+								if("AB+") typeRange = 6
+								if("O-") typeRange = 7
+								if("O+") typeRange = 8
+								if("AB-")
+									if(pick(1, 2) == 1)
+										typeRange = 9
+									else
+										typeRange = 10
+								else 
+									if(pick(1, 2, 3) == 1)
+										typeRange = 12
+									else
+										typeRange = 11
+						else
+							typeRange = rand(1,12)
+						var/datum/bioEffect/NEW = null
+						switch(typeRange)
+							if(2) NEW = new /datum/bioEffect/hemochromia/bronze()
+							if(3) NEW = new /datum/bioEffect/hemochromia/gold()
+							if(4) NEW = new /datum/bioEffect/hemochromia/lime()
+							if(5) NEW = new /datum/bioEffect/hemochromia/olive()
+							if(6) NEW = new /datum/bioEffect/hemochromia/jade()
+							if(7) NEW = new /datum/bioEffect/hemochromia/teal()
+							if(8) NEW = new /datum/bioEffect/hemochromia/cobalt()
+							if(9) NEW = new /datum/bioEffect/hemochromia/indigo()
+							if(10) NEW = new /datum/bioEffect/hemochromia/purple()
+							if(11) NEW = new /datum/bioEffect/hemochromia/violet()
+							if(12) NEW = new /datum/bioEffect/hemochromia/fuschia()
+							else NEW = new /datum/bioEffect/hemochromia/rust()
+
+						if(duplicateCheck == 0)
+							H.bioHolder.AddEffectInstance(NEW,1)
+						holder.RemoveEffect(src.id)
+						holder.RemovePoolEffect(src)
+					else
+						if (H.blood_volume < 500 && H.blood_volume > 0)
+							H.blood_volume += 2
+
+/datum/bioEffect/hemochromia/rust
+	name = "Hemochromia Type-R"
+	desc = "Causes the subject's blood cells to take on a scarlet coloration. Also slightly increases blood viscosity."
+	id = "hemochromia_rust"
+	msgGain = "You feel a bit less soft."
+	msgLose = "You feel lightheaded for a moment."
+	lockProb = 5
+	lockedChars = list("A")
+	occur_in_genepools = 0
+	research_level = 2
+	icon_state  = "hemochromia_rust"
+
+	OnAdd()
+		..()
+		blood_color_R = rand(254, 46)
+		blood_color_G = rand(0, 0)
+		blood_color_B = rand(0, 2)
```

Comment:
Why set the blood color in OnLife? Wouldn't just setting it once in OnAdd be enough?

Reply:
It would "regenerate" to the Hemochromia color for example after the subject ingests some colorful reagent. Otherwise after the blood color gets set by something else, the Hemochromia mutation will become completely useless.